internal/p2p/peermanager.go

```diff
@@ -589,6 +618,11 @@ func (m *PeerManager) Dialed(address NodeAddress) error {
 		return fmt.Errorf("peer %q was removed while dialing", address.NodeID)
 	}
 	now := time.Now().UTC()
+	if peer.Inactive {
+		m.metrics.PeersInactivated.Add(-1)
```

Comment:
Can we log an error here - there's something wrong if we see this happening

Reply:
I mean it shouldn't happen, logically, but if it does (because of some kind of race against the peer manager or peer store, or a really slow handshake,) then it's not a but at all... when this function is called we've successfully connected to a peer.

Reply:
Yeah I'm still not sure if we should be using this as a form of defensive programming to see if we have a race or some inconsistency

Reply:
One possible middle ground could be to add an additional metric for the bad case. That way the "safe default" still lets the system make progress, but we could trigger an alert on "some number of the bad thing happened". 

That can be a reasonable approach for issues that should not happen but which aren't serious enough to break a connection or fail the process.

Reply:
I think adding a new metric for a situation that we could also (arguably) panic for isn't great (and not super actionable for users. 

I guess, I'm of the thought that while this seems unlikely now, and is something that we could decide is an error, the "right thing" to do here is unambiguous, and we have all the information to handle this situation in the scope of this operation.